openstack/standalone-cinder/pkg/volumeservice/actions.go

```diff
@@ -109,6 +112,22 @@ func ConnectCinderVolume(vs *gophercloud.ServiceClient, volumeID string) (Volume
 	return rcv.ConnectionInfo, nil
 }
 
+// AttachCinderVolume marks the volume as attached in the cinder database.
+func AttachCinderVolume(vs *gophercloud.ServiceClient, volumeID string) error {
+	opts := volumeactions.AttachOpts{
```

Comment:
should we check volume status here? This could help avoid (but not eliminate) race condition
https://developer.openstack.org/api-ref/block-storage/v3/#attach-volume-to-a-server
```
Preconditions
Volume status must be available.
You should set instance_uuid or host_name.
```

Reply:
Actually the volume must be Available even for the reserve step.  It's why we do the polling there.